src/XRP/model/xrp-issued-currency.ts

```diff
@@ -19,20 +21,32 @@ export default class XrpIssuedCurrency {
     issuedCurrency: IssuedCurrencyAmount,
   ): XrpIssuedCurrency | undefined {
     const currency = issuedCurrency.getCurrency()
-    const xrpCurrency = currency && XrpCurrency.from(currency)
+    if (!currency) {
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'Issued currency protobuf does not contain valid `currency` field.',
+      )
+    }
+    const xrpCurrency = XrpCurrency.from(currency)
```

Comment:
The nice thing about what we're doing is that once all of these classes are throwing descriptive errors, we don't need to worry about surrounding this for checks as to whether the `XrpCurrency` object was successfully constructed from the `Currency` proto, because we know that `XrpCurrency.from` will already throw if that's the case.